apps/src/templates/census/CensusForm.jsx

```diff
@@ -31,159 +29,141 @@ export const censusFormPrefillDataShape = PropTypes.shape({
+}) => {
+  const [state, setState] = useState({
+    showFollowUp: false,
+    showPledge: false,
+    selectedHowMuchCS: [],
+    selectedTopics: [],
+    otherTopicsDesc: '',
+    schoolName: prefillData.schoolName ?? '',
+    schoolYear: initialSchoolYear,
+  });
+  const [submission, setSubmission] = useState({
+    name: prefillData.userName ?? '',
+    email: prefillData.userEmail ?? '',
+    role: prefillData.isTeacher ? 'TEACHER' : '',
+    country: prefillData.schoolCountry ?? 'United States',
+    hoc: '',
+    schoolName: prefillData.schoolName ?? '',
+    schoolCity: '',
+    schoolState: prefillData.schoolState ?? '',
+    schoolZip: prefillData.schoolZip ?? '',
+    schoolType: prefillData.schoolType ?? '',
+    afterSchool: '',
+    tenHours: '',
+    twentyHours: '',
+    otherCS: false,
+    followUpFrequency: '',
+    followUpMore: '',
+    acceptedPledge: false,
+    share: '',
+    optIn: '',
+  });
+  const [errors, setErrors] = useState({
+    invalidEmail: false,
+  });
+
+  useEffect(() => {
+    setState(prevState => ({
+      ...prevState,
+      showFollowUp:
+        submission.twentyHours === 'SOME' || submission.twentyHours === 'ALL',
+    }));
+  }, [submission.twentyHours]);
+
+  useEffect(() => {
+    setState(prevState => ({
+      ...prevState,
+      showPledge:
+        submission.role === 'TEACHER' || submission.role === 'ADMINISTRATOR',
+    }));
+  }, [submission.role]);
+
+  const handleChange = (field, event) => {
+    setSubmission(prevState => ({
+      ...prevState,
+      [field]: event.target.value,
+    }));
   };
 
-  constructor(props) {
-    super(props);
-
-    const prefillData = this.props.prefillData || {};
-
-    this.state = {
-      showFollowUp: false,
-      showPledge: false,
-      selectedHowMuchCS: [],
-      selectedTopics: [],
-      otherTopicsDesc: '',
-      schoolName: prefillData['schoolName'] || '',
-      schoolYear: this.props.initialSchoolYear,
-      submission: {
-        name: prefillData['userName'] || '',
-        email: prefillData['userEmail'] || '',
-        role: prefillData['isTeacher'] ? 'TEACHER' : '',
-        country: prefillData['schoolCountry'] || 'United States',
-        hoc: '',
-        schoolName: prefillData['schoolName'] || '',
-        schoolCity: '',
-        schoolState: prefillData['schoolState'] || '',
-        schoolZip: prefillData['schoolZip'] || '',
-        schoolType: prefillData['schoolType'] || '',
-        afterSchool: '',
-        tenHours: '',
-        twentyHours: '',
-        otherCS: false,
-        followUpFrequency: '',
-        followUpMore: '',
-        acceptedPledge: false,
-        share: '',
-        optIn: '',
-      },
-      errors: {
-        invalidEmail: false,
-      },
-    };
-  }
-
-  handleChange = (field, event) => {
-    this.setState(
-      {
-        submission: {
-          ...this.state.submission,
-          [field]: event.target.value,
-        },
-      },
-      this.checkShowFollowUp
```

Comment:
this callback (line 113 as well) is moved to a useEffect, since the useState setter does not have a callback argument